### PR TITLE
feat: introduce configurable resource limits for flagd sidecar

### PR DESCRIPTION
**Pull request by skyerus**

### CODE REVIEW

**Comment by beeme1mr** [comment]
It would be great to leverage some benchmark metrics to help define reasonable default resource limits.

There's an open issue in the flagd repo requesting this information.

https://github.com/open-feature/flagd/issues/212

File: main.go [hunk 1]
```diff
@@ -60,10 +63,12 @@ func main() {
 
 	flag.StringVar(&metricsAddr, "metrics-bind-address", ":8080", "The address the metric endpoint binds to.")
 	flag.StringVar(&probeAddr, "health-probe-bind-address", ":8081", "The address the probe endpoint binds to.")
-	flag.BoolVar(&verbose, "verbose", true, "Disable verbose logging (default: true)")
+	flag.BoolVar(&verbose, "verbose", true, "Disable verbose logging")
 	flag.BoolVar(&enableLeaderElection, "leader-elect", false,
 		"Enable leader election for controller manager. "+
 			"Enabling this will ensure there is only one active controller manager.")
+	flag.StringVar(&flagDCpuLimit, "flagd-cpu-limit", "0.5", "flagd CPU limit, in cores. (500m = .5 cores)")
```
Comment (skyerus) [inline main.go:70]: will comment a link to the motivation behind these default values once https://github.com/open-feature/flagd/pull/225/files has hit main

File: main.go [hunk 2]
```diff
@@ -111,6 +128,12 @@ func main() {
 	//+kubebuilder:scaffold:builder
 	hookServer := mgr.GetWebhookServer()
 	hookServer.Register("/mutate-v1-pod", &webhook.Admission{Handler: &webhooks.PodMutator{
+		FlagDResourceRequirements: corev1.ResourceRequirements{
```
[the remark this answers is not in the document]
Comment (skyerus) [reply main.go:131]: I agree that having a minimum memory/cpu usage is better than it defaulting to the maximum, I'll implement this.
I am curious though, why would having the minimum usage the same as the maximum make the pod unstable?

[the remark this answers is not in the document]
Comment (Kavindu-Dodan) [reply main.go:151]: I have seen pod restarts due to "OOMKilled" and the explanation is that when pod gets more and more load, it is allowed to reach the "request" limit. However, if pod tries to consume a bit more resources, it overshoots the "limit" (as they are the same). So having a buffer is better.

[the remark this answers is not in the document]
Comment (beeme1mr) [reply main.go:151]: @AlexsJones do you have any thoughts on this?

[the remark this answers is not in the document]
Comment (skyerus) [reply main.go:151]: Thanks for the explanations, that makes sense. I've included request resources.

**Review by Kavindu-Dodan — APPROVED** [review]

**Review by toddbaert — APPROVED** [review]

File: chart/templates/rendered.yaml [hunk 3]
```diff
@@ -812,6 +812,8 @@ spec:
         - --health-probe-bind-address=:8081
         - --metrics-bind-address=127.0.0.1:8080
         - --leader-elect
+        - --flagd-cpu-limit=0.5
```
Comment (james-milligan) [inline chart/templates/rendered.yaml:815]: We should pass these values as value injections with helm, you can overwrite the values through the overlay in `/config/overlays/helm` with a `patchesStrategicMerge`. Which will let developers update these values via the `values.yaml` file.
Happy to introduce this change in a separate PR

Comment (james-milligan) [reply chart/templates/rendered.yaml:815]: I will introduce this change in https://github.com/open-feature/open-feature-operator/issues/271

**Review by james-milligan — APPROVED** [review]

**Review by AlexsJones — APPROVED** [review]

**Pull request merged**